src/memex/resources.py

```diff
@@ -16,17 +16,19 @@ def __getitem__(self, id):
         annotation = storage.fetch_annotation(self.request.db, id)
         if annotation is None:
             raise KeyError()
-        return AnnotationResource(self.request, annotation)
+
+        group_service = self.request.find_service(IGroupService)
```

Comment:
Nice. The division between `AnnotationResourceFactory` and `AnnotationResource` means that the latter is easily usable on its own (i.e. without a request.)